common/persistence/client/factory.go

```diff
@@ -189,11 +199,20 @@ func (f *factoryImpl) NewNamespaceReplicationQueue() (p.NamespaceReplicationQueu
 	if f.metricsClient != nil {
 		result = p.NewQueuePersistenceMetricsClient(result, f.metricsClient, f.logger)
 	}
-
+	result = p.NewQueuePersistenceRetryableClient(result, retryPolicy, IsPersistenceTransientError)
 	return p.NewNamespaceReplicationQueue(result, f.serializer, f.clusterName, f.metricsClient, f.logger)
 }
 
 // Close closes this factory
 func (f *factoryImpl) Close() {
 	f.dataStoreFactory.Close()
 }
+
+func IsPersistenceTransientError(err error) bool {
+	switch err.(type) {
+	case *serviceerror.Unavailable:
+		return true
```

Comment:
under what condition does persistence layer return Unavailable error?

Reply:
I couldn't find any usage for cassandra, but for sql, it's returned here: https://github.com/temporalio/temporal/blob/master/common/persistence/sql/common.go#L66

I removed the resource exhausted error from the existing [IsPersistenceTransientError](https://github.com/temporalio/temporal/blob/master/common/util.go#L234) implementation, so only Unavailable error is remained here.